### PR TITLE
VPN UI improvements part 4 - location and data volume sections

### PR DESCRIPTION
**Pull request by samsymons**
Task/Issue URL: https://app.asana.com/0/414235014887631/1206747946345658/f
Tech Design URL:
CC:

**Description**:

This PR updates the location section and adds the new data volume section.

BSK PR: https://github.com/duckduckgo/BrowserServicesKit/pull/722

**Steps to test this PR**:
1. Run the branch and check that the location label matches your setting
2. Select another option and make sure it updates
3. Select the Nearest Available setting and then connect NetP
4. Check that the UI shows "Your Location (Nearest)"
5. Check that when you're connected, the Data Volume row updates regularly
6. Exit the screen and then enter it again, and check that the Data Volume row is what you expect
7. Disconnect and reconnect
8. Check that Data Volume has been reset to 0 KB for each value

**Copy Testing**:

* [ ] Use of correct apostrophes in new copy, ie `’` rather than `'`

**Orientation Testing**:

* [ ] Portrait
* [ ] Landscape

**Device Testing**:

* [ ] iPhone SE (1st Gen)
* [ ] iPhone 8
* [ ] iPhone X
* [ ] iPhone 14 Pro
* [ ] iPad

**OS Testing**:

* [ ] iOS 14
* [ ] iOS 15
* [ ] iOS 16

**Theme Testing**:

* [ ] Light theme
* [ ] Dark theme

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: DuckDuckGo.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -10043,8 +10047,8 @@
 			isa = XCRemoteSwiftPackageReference;
 			repositoryURL = "https://github.com/DuckDuckGo/BrowserServicesKit";
 			requirement = {
-				kind = exactVersion;
-				version = 122.1.0;
+				branch = "sam/vpn-ui-improvements";
+				kind = branch;
```
Comment (samsymons) [inline DuckDuckGo.xcodeproj/project.pbxproj:10051]: This is the only PR of the batch that needs a BSK update, I'll assign you a separate task for this on Wednesday.

File: DuckDuckGo/NetworkProtectionFAQView.swift [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        self.webView = NetworkProtectionWebView(configuration: .nonPersistent())
+        self.webView.load(url: Constants.frequentlyAskedQuestionsURL)
+    }
+
+    var body: some View {
+        VStack {
+            webView
+                .ignoresSafeArea()
+        }
+        .navigationTitle("DuckDuckGo VPN FAQ")
```
Comment (samsymons) [inline DuckDuckGo/NetworkProtectionFAQView.swift:44]: Some copy changes here need to be cleaned up and moved into UserText.

File: DuckDuckGo/NetworkProtectionStatusView.swift [hunk 3]
```diff
@@ -27,7 +27,6 @@ struct NetworkProtectionStatusView: View {
     @Environment(\.colorScheme) var colorScheme
 
     @StateObject public var statusModel: NetworkProtectionStatusViewModel
-    @State private var isFeedbackFormActive = false
```
Comment (samsymons) [inline DuckDuckGo/NetworkProtectionStatusView.swift:30]: No longer need state for this, the FAQ screen is just another navigation link.

File: DuckDuckGo/NetworkProtectionStatusView.swift [hunk 4]
```diff
@@ -126,16 +129,53 @@ struct NetworkProtectionStatusView: View {
...
+                NavigationLink(destination: NetworkProtectionVPNLocationView()) {
+                    NetworkProtectionLocationItemView(title: locationAttributedString, imageName: nil)
+                }
+            } else {
+                let imageName = statusModel.preferredLocation.isNearest ? "VPNLocation" : nil
+                var nearestLocationAttributedString: AttributedString {
+                    var attributedString = AttributedString(statusModel.preferredLocation.title)
+                    attributedString.foregroundColor = .init(designSystemColor: .textPrimary)
+                    return attributedString
+                }
```
Comment (quanganhdo) [inline DuckDuckGo/NetworkProtectionStatusView.swift:155]: Nice, didn't know I could do this. Normally I use `let xxx = { ... }()`

File: DuckDuckGo/NetworkProtectionStatusView.swift [hunk 5]
```diff
@@ -126,16 +129,53 @@ struct NetworkProtectionStatusView: View {
...
+
                 NavigationLink(destination: NetworkProtectionVPNLocationView()) {
-                    NetworkProtectionServerItemView(title: UserText.netPStatusViewLocation, value: location)
+                    NetworkProtectionLocationItemView(title: nearestLocationAttributedString, imageName: imageName)
                 }
             }
+        } header: {
+            Text(statusModel.isNetPEnabled ? "Connected Location" : "Selected Location").foregroundColor(.init(designSystemColor: .textSecondary))
```
Comment (quanganhdo) [inline DuckDuckGo/NetworkProtectionStatusView.swift]: Are these going to be localized?

**Review by quanganhdo — APPROVED** [review]
The only issue I ran into is that the FAQ web view retains its history, so going back from that to VPN Settings and reopening the FAQ sends you to the most recent page, not the FAQ homepage. 

Otherwise, it works great.

File: DuckDuckGo/NetworkProtectionStatusView.swift (hunk 5 above)
Comment (samsymons) [reply DuckDuckGo/NetworkProtectionStatusView.swift]: Good find, I thought I got all these already, these should definitely be moved into UserText.

**Pull request merged**